src/TypeScript/TsConfig/TsConfigProvider.cs

```diff
@@ -77,25 +77,33 @@ public async Task<string> GetConfigForFile(string sourceFilePath, CancellationTo
         {
             foreach (var tsConfigFilePath in candidateTsConfigs)
             {
-                if (visited.Contains(tsConfigFilePath))
+                var tsConfigToCheck = tsConfigFilePath;
+                var isDirectory = !Path.HasExtension(tsConfigToCheck);
```

Comment:
Paths can contain full stops, in which case `Path.HasExtension` will return the wrong result.

`Directory.Exists` returns the correct result.

Reply:
Ah, you're right. I should've used `Path.HasExtension(Path.GetFullPath(tsConfigToCheck))`, because then a directory would've always ended with "\\". 
`Directory.Exists` -- it doesn't have to exist, e.g. if you input a wrong directory name in the external tsconfig.json. The ts compiler will complain, but it doesn't stop us from running the analysis and failing with:
`Failed to process 'C:/Users/Rita/source/repos/NodejsConsoleApp3/NewFolder1/a': Cannot read file 'C:/Users/Rita/source/repos/NodejsConsoleApp3/NewFolder1/a'.`
Overall I think it doesn't matter, I'll change it to directory.

Reply:
Hhmm - what about relative paths? Using Path.HasExtension with a canonical path might work better than Directory.Exists in that case.

Reply:
Eslint-bridge returns us full file paths, even if the tsconfig contains a relative path. So we're ok with that.

Reply:
Overall I think if the user input something wrong (i.e. non existing directory, or a directory that doesn't have a tsconfig.json), the analysis will likely fail either way, and so would the build itself. So as long as we correctly support directories in general and don't fail with an ugly exception, I think it should be ok.

Reply:
It's probably worth adding an assertion that the paths returned from eslintbridge are rooted i.e. to document and validate our assumptions about the behaviour of an external system.